common/src/main/java/com/microsoft/identity/common/internal/fido/IFidoManager.kt

```diff
@@ -22,16 +22,15 @@
 // THE SOFTWARE.
 package com.microsoft.identity.common.internal.fido
 
-import android.webkit.WebView
-import com.microsoft.identity.common.internal.ui.webview.challengehandlers.IChallengeHandler
-
 /**
- * Abstract class that handles a FidoChallenge.
+ * Representation of a manager that handles interactions with a passkey provider (usually through an API).
  */
-abstract class AbstractFidoChallengeHandler
-/**
- * Constructs an AbstractFidoChallengeHandler.
- * @param webView current WebView.
- */ (
-    val webView: WebView
-) : IChallengeHandler<IFidoChallenge, Void>
+interface IFidoManager {
+    /**
+     * Interacts with the FIDO credential provider and returns an assertion.
+     * @param challenge AuthFidoChallenge received from the server.
+     * @return assertion
+     * @throws Exception
+     */
+    suspend fun authenticate(challenge: AuthFidoChallenge): String
```

Comment:
IFidoChallenge?

Reply:
IFidoChallenge currently only has AuthFidoChallenge implementing it, but RegFidoChallenge could be an object in the future (original plan was to implement both registration and authentication, but the plans for native registration have been dropped for the time being). So my thinking was to have an authenticate method which only takes AuthFidoChallenges, and a register method which only takes RegFidoChallenges.

Unless I'm missing some fundamental design principle here.

Reply:
They wouldn't be used interchangeably, but the two of them do share many fields (such as the ones defined in IFidoChallenge). This was the reason why I created an interface.